packages/gasket-plugin-intl/lib/fs-utils.js

```diff
@@ -1,57 +1,62 @@
 const fs = require('fs-extra');
 const path = require('path');
 
-class FsUtils {
+/**
+ * Tuple of package name and path
+ * @typedef {[string, string]} SrcPkgDir
+ * @private
+ */
 
-  /**
-   * Check if path is module. It will look for a package.json file.
-   *
-   * @param {string} targetDir - File path
-   * @returns {boolean} result
-   */
-  async isModule(targetDir) {
-    try {
-      const stats = await fs.stat(path.join(targetDir, 'package.json'));
-      return !!stats;
-    } catch (e) {
-      return false;
-    }
+/**
+ * Check if path has package.json file and return the name.
+ *
+ * @param {string} targetDir - File path
+ * @returns {string|undefined} result
+ */
+async function packageName(targetDir) {
+  try {
+    const pkg = await fs.readJson(path.join(targetDir, 'package.json'));
+    return pkg.name;
+  } catch (e) {
+    // skip;
   }
+}
 
-  /**
-   * Find all directories under target dir, recursively
-   *
-   * @param {string} parentDir - Path to parent directory
-   * @param {string[]} dirList - List of full paths
-   * @returns {string[]} directories - List of full paths
-   */
-  async getDirectories(parentDir, dirList = []) {
-    const files = await fs.readdir(parentDir);
-    for (const file of files) {
-      const filePath = path.join(parentDir, file);
-      if ((await fs.stat(filePath)).isDirectory()) {
-        if (await this.isModule(filePath)) {
-          dirList.push(filePath);
-        } else {
-          dirList.concat(await this.getDirectories(path.join(filePath, '/'), dirList));
-        }
+/**
+ * Find all directories under target dir, recursively
+ *
+ * @param {string} parentDir - Path to parent directory
+ * @param {SrcPkgDir[]} dirList - List of full paths
+ * @returns {AsyncGenerator<SrcPkgDir>} source package directories
+ */
+async function *getPackageDirs(parentDir, dirList = []) {
+  const files = await fs.readdir(parentDir);
+  for (const file of files) {
+    const filePath = path.join(parentDir, file);
+    if ((await fs.stat(filePath)).isDirectory()) {
+      const pkgName = await packageName(filePath);
+      if (pkgName) {
+        yield await [pkgName, filePath];
```

Comment:
You shouldn't need the await.

```suggestion
        yield [pkgName, filePath];
```